pinot-query-planner/src/test/java/org/apache/pinot/query/QueryEnvironmentTestBase.java

```diff
@@ -202,4 +205,10 @@ public String toString() {
       }
     }
   }
+
+  // temporary function to strip the Physical plan from the explain query plan.
+  // physical plan is present for the explain queries by default.
```

Comment:
Added this temporary function to strip the physical plan from the output. 

@walterddr question, do we want physical and logical plans per the default Calcite default implementation, or do we want to change it? I will remove this temp function based on the feedback.

Reply:
can you share an example explain results on both approaches?

Reply:
just updated the queries with the physical plan.

Reply:
hmm. I am confused. do you mean `EXPLAIN PLAN FOR ...` by default also create the physical plan? that doesn't seem right

Reply:
yes,  by default it's setting `SqlExplain.Depth.PHYSICAL`. 
Please check class `org.apache.calcite.sql.parser.impl.SqlParserImpl` method `ExplainDepth()`, line number 5013. It's a generated file so I am missing the link here. I did check the calcite project, it's the same for Babel and a couple of other parsers impl. 

That's why I was asking what type of default behaviour we want.

Reply:
Depth is an enum with 3 values
```
  public static enum Depth implements Symbolizable {
    TYPE,
    LOGICAL,
    PHYSICAL;

    private Depth() {
    }
  }
```
it is 3 different enum so the result should be just 3 different format and should not mix and match

Reply:
Right, so default value is PHYSICAL.

All the explain queries `without implementation` or `with type` will result the physical plan. Is it good?

Reply:
ok now I understand the problem.. sorry for the confusion, it seems like we should set the default to LOGICAL instead of PHYSICAL

Reply:
I was going over to see how we can set the default to LOGICAL, which requires some hacks. There are two options I can think of,
1. Maintaining `Parser.jj` copy in Pinot instead of extracting it from the calcite.jar. The parser has the [code](https://github.com/apache/calcite/blob/main/core/src/main/codegen/templates/Parser.jj#L1455) for setting the default value as Physical. I tried to put a similar function in our `parserImpls.ftl` to override the function, but it seems like `fmpp` has no such overriding functionality, and copying both methods failed while generating a parser class file.
2. Before Parsing the `Explain` SQL query, check whether the `with` option is provided. If not, add `WITHOUT IMPLEMENTATION` in the query. 

Please suggest a preferable option you have in mind or among the above two. I will try to dig more if I can get some other trick. Thanks

Reply:
hmm. at this stage i think it is easier to not follow the calcite standard. and create a extension `SqlPhysicalExplain` node and a special parsing logic in `parserImpls.ftl` to directly use a different logic. WDYT?

Reply:
Thanks for mentioning the option and code; this is much better than the other two options I have suggested. Updated the changes.

Reply:
thanks I will take another look later